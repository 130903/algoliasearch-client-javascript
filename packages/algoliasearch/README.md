# algoliasearch

> Universal JavaScript API client for Algolia search API.

See https://github.com/algolia/algoliasearch-client-js for the library documentation.

See https://www.algolia.com/doc/rest for the underlying REST API documentation.

See https://www.algolia.com/doc for the main Algolia documentation.
