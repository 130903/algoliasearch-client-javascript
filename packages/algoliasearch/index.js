import debug from 'debug';
debug.useColors = false;

export {default} from './src/algoliasearchNodejs.js';
