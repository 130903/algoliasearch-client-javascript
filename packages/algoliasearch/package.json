{
  "name": "algoliasearch",
  "version": "4.0.0-alpha",
  "description": "Universal JavaScript API client for Algolia search API.",
  "repository": "https://github.com/algolia/algoliasearch-client-js/tree/master/packages/algoliasearch-lite",
  "keywords": [
    "algolia",
    "search",
    "api client"
  ],
  "homepage": "https://github.com/algolia/algoliasearch-client-js",
  "bugs": "https://github.com/algolia/algoliasearch-client-js/issues",
  "author": "Algolia SAS (https://www.algolia.com)",
  "license": "MIT"
}
