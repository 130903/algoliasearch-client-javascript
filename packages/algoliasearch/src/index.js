// @flow

export { default as initClient } from './constructors/initClient';
export { default as initIndex } from './constructors/initIndex';
export { default as initPlaces } from './constructors/initPlaces';
export { default as generateSecuredApiKey } from './generateSecuredApiKey';
import { default as deprecateDefaultImport } from './deprecateDefaultImport';
export default deprecateDefaultImport;
