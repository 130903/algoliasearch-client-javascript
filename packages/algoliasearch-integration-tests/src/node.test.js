import { initClient } from 'algoliasearch';

// todo: figure out how to avoid running this test in regular mode when destruct
jest.unmock('algoliasearch-http-requester');

it('can initialize a client', () => {
  const client = initClient({
    appId: 'latency',
    apiKey: '6be0576ff61c053d5f9a3225e2a90f76',
  });
  expect(
    client.search({ requests: [{ indexName: 'bestbuy', query: 'a' }] })
  ).resolves.toMatchSnapshot();
});
