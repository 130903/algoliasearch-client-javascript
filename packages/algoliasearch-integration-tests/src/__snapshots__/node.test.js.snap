// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can initialize a client 1`] = `Object {}`;
